### PR TITLE
sol2: conan v2 support

### PR DESCRIPTION
**Pull request by StellaSmith**
Requires:
- [x] Lua conan v2 support

### CODE REVIEW

**Comment by StellaSmith** [comment]
@ericLemanissier @prince-chrismc
Linter seems to be failing. Does it not like that I deleted those files?

**Comment by prince-chrismc** [comment]
Yes https://github.com/conan-io/conan-center-index/actions/runs/4060268796/jobs/6989189241#step:8:24 🤔

**Comment by StellaSmith** [comment]
Linters passed first try.
 Success!

File: recipes/sol2/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                self.requires("lua/5.4.4")
+        elif self.options.with_lua == "luajit":
+            self.requires("luajit/2.1.0-beta3")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    @property
+    def _compilers_minimum_version(self):
+        # TODO: MSVC?
```
Comment (prince-chrismc) [inline recipes/sol2/all/conanfile.py:46]: You can steal it from the template 

https://github.com/conan-io/conan-center-index/blob/d23f5b91aef60cdf22b927a183327de0a2b0bd68/docs/package_templates/cmake_package/all/conanfile.py#L80-L81

File: recipes/sol2/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        if self.version < "3.0.0":
+            return {
+                "Visual Studio": "14",
```
Comment (prince-chrismc) [inline recipes/sol2/all/conanfile.py:49]: https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B#Internal_version_numbering

Should be 190 with new numbers

```suggestion
```

File: recipes/sol2/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            return {
+                "Visual Studio": "15.7" if Version(self.version) < "3.3.0" else "16",
+                "gcc": "7",
+                "clang": "6",
+                "apple-clang": "10",
+            }
+
+    @property
+    def _min_cppstd(self):
+        if self.version < "3.0.0":
```
Comment (prince-chrismc) [inline recipes/sol2/all/conanfile.py:64]: I am actually not sure what the answer is on this one... but putting it the team lol

```suggestion
        if Version(self.version) < "3.0.0":
```

**Review by prince-chrismc — COMMENTED** [review]
Tiny suggestions but nothing was a blocker 🤔

File: recipes/sol2/all/conanfile.py (hunk 3 above)
Comment (memsharded) [reply recipes/sol2/all/conanfile.py:64]: Yes, explicitly converting to version would be better. The ``conan.tools.scm`` ``Version`` does a ``str()`` internally, so even if ``self.version`` is a Version in 1.X, it should be safe.

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Build failed with commits not being sync'd :/ can you please merge master to start a new one ?

**Review by uilianries — DISMISSED** [review]
LGTM

**Review by TheClonerx — DISMISSED** [review]

**Comment by StellaSmith** [comment]
Should I also set `transitive_headers` & `transitive_libs` to true for the Lua & LuaJIT requirements?

**Review by prince-chrismc — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by TheClonerx — APPROVED** [review]

**Pull request merged**